### DISCUSSION TITLE
Github Pages doesnt display japanese characters correctly

### DISCUSSION THREAD

**Discussion by xl3lackout** [community/community#22034] [Other Feature Feedback, Questions, & Ideas]
<p>Ive looked around for solutions but I havent been able to find much discussion on the topic. Im essentially hosting a wiki translation project on github pages, and untranslated sections are appearing as invalid characters. It displays fine when opening the html directly from file, and checking the files themselves on github show that they still contain the kanji, but looking at the deployment just breaks them. Any info would be very appreciated</p>

**Comment by yoannchaudet** [comment]
<p>Hello and welcome here!</p>
<p>I suggest you double check the encoding of your raw HTML files.</p>
<p>They are all mingled in the repository already, like here for instance: <a href="https://github.com/xl3lackout/MHFZ-Ferias-English-Project/blob/90e26e1bf48bb9d774cf751f77afeb9fa3aea59d/index.html#L12" class="inline-onebox">MHFZ-Ferias-English-Project/index.html at 90e26e1bf48bb9d774cf751f77afeb9fa3aea59d · xl3lackout/MHFZ-Ferias-English-Project · GitHub</a></p>

**Comment by xl3lackout** [comment]
<p>Sorry for not mentioning, but the sites were downloaded using a tool that already didnt support kanji, ive had to redownload portions using a different tool and translate them, so unfinished sections still have intelligible wording. Theres a lot of preserved kanji under the sozai folder, but if you go to the Item Index section of the website, which is what sozai covers, thats where a lot of the kanji gets incorrectly displayed. My apologizes, and I appreciate the response nonetheless.</p>

**Comment by xl3lackout** [comment]
<p>Heres a good comparison on what it looks like when looking at it from file, and looking at the github pages build.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/9/6/96182a41946d986cc8f5203a9692643831425340.png" data-download-href="/uploads/short-url/lpNoqrBX1I7vHAgJe99HX9FLOKs.png?dl=1" title="file" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/45904365/181068381-3177e33b-1b08-4250-9548-acb21f98a02e.png" alt="file" data-base62-sha1="lpNoqrBX1I7vHAgJe99HX9FLOKs" width="690" height="470" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/6/96182a41946d986cc8f5203a9692643831425340_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">file</span><span class="informations">1136×775 44 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div><br>
<div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/2/a2ca809ae40d561bfbcb6b4394aa6b54886ec32c.png" data-download-href="/uploads/short-url/ne7cZpN5fw26twJk0PzInRV5wWo.png?dl=1" title="github" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/45904365/181068382-fb162d86-e43f-4505-a8a1-f67504e2f13c.png" alt="github" data-base62-sha1="ne7cZpN5fw26twJk0PzInRV5wWo" width="690" height="480" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/2/a2ca809ae40d561bfbcb6b4394aa6b54886ec32c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">github</span><span class="informations">1155×804 47.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by yoannchaudet** [comment]
<p>I suggest you take the following steps:</p>
<ol>
<li>Commit a <code>.nojekyll</code> empty file at the root of your repository (without this, your site builds with <a href="https://jekyllrb.com/">Jekyll</a> and has a <a href="https://jekyllrb.com/docs/configuration/options/">default encoding</a> you are not expecting)</li>
<li>
<a href="https://www.w3.org/International/questions/qa-html-encoding-declarations">Update your html files to specify which character encoding you need</a> (likely <code>utf-8</code> here)</li>
<li>Clean up your files slowly :slight_smile:
</li>
</ol>

**Comment by xl3lackout** [comment]
<p>I tried what you suggested, and it didnt work at first. But i did some more research and found that i had to manually change the encoding of the file itself from Shift-JIS to UTF-8 using notepad++. After that it displayed correctly. Thank you for your assistance!</p>
